posts/2019/11/Variable-Explorer.md

```diff
@@ -0,0 +1,73 @@
+To expose this new functionality we also decided to turn off the Variable Explorer option called “Exclude unsupported types” and introduce a new one called “Exclude callables and modules” (turned on by default) to display a much larger fraction of objects after a file or cell execution.
+
+![Exclude callables modules](/images/spyder-variable-explorer/exclude-callables-modules.png)
+
+Finally, we added a context-menu action to allow using the object explorer viewer for all objects, regardless if they have a default viewer or not.
+
+![View object explorer](/images/spyder-variable-explorer/view-object-explorer.png)
+
+## Multi-index support in the dataframe viewer
+
+One of the first enhancements we did to the Variable Explorer in Spyder 4 was adding support for multi-dimensional indexes to its dataframe viewer. In Spyder 3 the support was very limited, which made inspecting those indexes a painful experience:
+
+![Multi-index support](/images/spyder-variable-explorer/multi-index-support.png)
+
+For Spyder 4 we took advantage of the work done by Scott Hansen ([@firecat53](https://github.com/firecat53)) and Yuri D'Elia ([@wavexx](https://github.com/wavexx)) in their [gtabview](https://github.com/TabViewer/gtabview) project for this very purpose. The main element we took from it was its improved management of column and table headings, which allowed us to display the index shown above in a much nicer way:
+
+![Table headings](/images/spyder-variable-explorer/table-headings.png)
+
+## Fuzzy filtering of variables
+
+This version also includes the ability to filter variables in the current namespace to focus only on the ones you are interested in. This filtering is done in a fuzzy way by matching any letter of text entered in the search field with the type and name of all available variables.
+To access this functionality you need to press the keyboard shortcut `Ctrl+F` when the Variable Explorer has focus or by pressing the magnifying glass icon on top,
+
+
+![Filter variables](/images/spyder-variable-explorer/filter-variables.png)
+
+To remove the filtering, simply press `Esc` or `Ctrl + F` when the Variable Explorer is focused, or the magnifying glass icon again.
+
+## Refresh while code is running
+
+Finally, we restored the ability to refresh the Variable Explorer while code is running in the console. This was lost in Spyder 3.2, when we removed the old and unmaintained Python console. It’s coming back thanks to the fantastic work done by Quentin Peter ([@impact27](https://github.com/impact27)), who completely re-architectured the way the Spyder interface talks to the Jupyter kernels in charge of running code in our IPython console, by using [Jupyter Comms](https://jupyter-client.readthedocs.io/en/stable/messaging.html#custom-messages).
+
+![Refresh](/images/spyder-variable-explorer/refresh.png)
+
+This can be achieved by by pressing the refresh button on the Variable Explorer toolbar, or the the shortcut `Ctrl+R` when it has focus.
+
```

Comment:
Add mention of set support, and brief roundup of a few other smaller improvements

```suggestion

## Full support for sets

In Spyder 3, the Variable Explorer could only builtin Python sets as arbitrary objects, making it very difficult for users to browse and interact with them. In Spyder 4, you can now viewing sets just like lists, as well perform as various operations on them.

![Screenshot of Spyder Variable Explorer UI, showing a Python set displayed in a native viewer](/images/spyder-variable-explorer/set-support.png)

## UI enhancements and more

Finally, beyond the headline features, we've added numerous smaller improvements to make the Variable Explorer easier and more efficient to use. These include support for custom index names, better and more efficient automatic resizing of column widths, support for displaying Pandas Indices, tooltips for truncated column headers, and more.

```

Reply:
I agree with this, but a *show* is missing:

> In Spyder 3, the Variable Explorer could only **show** builtin Python sets as ...

@CAM-Gerlach, please update your suggestion so that @juanis2112 can apply it.

Reply:
> ... as well **as** perform ~~as~~ ...

Reply:
Yep and yep, fixed, and also fixed another similarly missing word.